snapshot/snapshotfs/upload.go

```diff
@@ -729,7 +729,20 @@ func (u *Uploader) Upload(
 		Source: sourceInfo,
 	}
 
-	u.Progress.UploadStarted()
+	maxPreviousTotalFileSize := int64(0)
+	maxPreviousFileCount := 0
+
+	for _, m := range previousManifests {
+		if s := m.Stats.TotalFileSize; s > maxPreviousTotalFileSize {
+			maxPreviousTotalFileSize = s
+		}
+
+		if s := m.Stats.TotalFileCount; s > maxPreviousFileCount {
+			maxPreviousFileCount = s
+		}
+	}
+
```

Comment:
Would it make sense to make these values larger than the previous max by some factor?
The question is what that factor should be. At some point, for very large snapshots, a fraction of the existing snapshot size becomes much larger than a reasonable ingest / source growth rate.
Anyway, just a random idea, not even sure it makes sense.